referenceframe/frame_system.go

```diff
@@ -753,3 +755,43 @@ func TopologicallySortParts(parts []*FrameSystemPart) ([]*FrameSystemPart, error
 	}
 	return topoSortedParts, nil
 }
+
+func frameSystemsAlmostEqual(fs1, fs2 *FrameSystem, epsilon float64) (bool, error) {
+	if fs1.Name() != fs2.Name() {
+		return false, nil
+	}
+
+	worldFrameEquality, err := framesAlmostEqual(fs1.World(), fs2.World(), epsilon)
+	if err != nil {
+		return false, err
+	}
+	if !worldFrameEquality {
+		return false, nil
+	}
+
+	if !reflect.DeepEqual(fs1.parents, fs2.parents) {
+		return false, nil
+	}
+
+	seenFrames := map[string]struct{}{}
+	for frameName, frame := range fs1.frames {
+		frame2, ok := fs2.frames[frameName]
+		if !ok {
+			return false, nil
+		}
+		frameEquality, err := framesAlmostEqual(frame, frame2, epsilon)
+		if err != nil {
+			return false, err
+		}
+		if !frameEquality {
+			return false, nil
+		}
+		seenFrames[frameName] = struct{}{}
+	}
+	for _, name := range fs2.FrameNames() {
+		if _, ok := seenFrames[name]; !ok {
+			return false, nil
+		}
+	}
```

Comment:
I don't think we need to collect `seenFrames`.

All we need to do is check that `len(fs1.frames) == len(fs2.frames)` before going into the comparison.

What do you think?

Reply:
sure